lib/web_ui/lib/src/engine/embedder.dart

```diff
@@ -180,102 +133,32 @@ class FlutterViewEmbedder {
       '$defaultFontStyle $defaultFontWeight ${defaultFontSize}px $defaultFontFamily';
 
   void reset() {
-    final bool isWebKit = browserEngine == BrowserEngine.webkit;
-
-    _styleElement?.remove();
-    _styleElement = createDomHTMLStyleElement();
-    _resourcesHost?.remove();
-    _resourcesHost = null;
-    domDocument.head!.append(_styleElement!);
-    final DomCSSStyleSheet sheet = _styleElement!.sheet! as DomCSSStyleSheet;
-    applyGlobalCssRulesToSheet(
-      sheet,
-      browserEngine: browserEngine,
-      hasAutofillOverlay: browserHasAutofillOverlay(),
-    );
-
-    final DomHTMLBodyElement bodyElement = domDocument.body!;
-
-    bodyElement.setAttribute(
+    _applicationDom.setHostAttribute(
       'flt-renderer',
       '${renderer.rendererTag} (${FlutterConfiguration.flutterWebAutoDetect ? 'auto-selected' : 'requested explicitly'})',
     );
```

Comment:
I think `flt-renderer` and `flt-build-mode` are inherently global, so we can just stick them in the `<body>` tag. `ApplicationDom` shouldn't be concerned with them.

Reply:
They're global to the flutter application, but not global to the users' web application in which we're being embedded.

Our goal should be to *not* modify anything outside of the hostElement, because we don't know how the users' app is going to react to that.

Reply:
True, but for things that do not interact with anything else (like specifically these two attributes), I wouldn't be too worried. We already ask the developer to add `<script src="flutter.js"></>` globally, so we can't avoid altering global scope completely.

Having said that, my main concern is about these expanding the API surface of `ApplicationDom`. We can still keep these attributes local to the glasspane by having `ApplicationDom` set them on itself.

Reply:
> my main concern is about these expanding the API surface of `ApplicationDom`.

I figured this was the case, and moved them to be private for each implementation and into their init methods, without being part of the core API. The *values* that are applied to the DOM are passed from the outside, however.